Search, One-page forms and User mgmt section review (#10)
ezsystems/commerce-documentation: docs/guide/search/search_cookbook/custom_search.md

        $eshopQuery = new EshopQuery();
        $eshopQuery->addCondition(new SearchTermCondition(array(
            'searchTerm' => 'DECLINE',
            'fieldRestrictions' => 'ses_product_ses_discontinued_value_s'
        )));
  
        // 3. Add visibility
        $eshopQuery->addCondition(new VisibilityCondition(array('visibility' => 0)));
  
        // 4.b Add boosting (eZ Content)
        $boosts = array(
            'ses_name' => 50,
            'ses_sku' => 100
            'ses_intro' => 20
            'ses_short_description' => 2
            'ses_long_description' => 1.5
            'ses_manufacturer' => 2
        );
        // 4.c Add boosting (Econtent)
        $boosts = array(
            'text' => 1,
            'ses_product_ses_name_value_t' => 50,
            'ses_product_ses_sku_value_s' => 100
            'ses_product_ses_intro_value_t' => 20
            'ses_product_ses_short_description_value_t' => 2
            'ses_product_long_description_value_t' => 1.5
            'ses_product_ses_manufacturer_value_s' => 2
        );
        $eshopQuery->addCondition(new FieldBoosting(array('boost' => $boosts)));
  

>         // 5. Add offset and limit.

Reviewer: ```suggestion
        // 5. Add offset and limit
```

        $eshopQuery->setOffset(0);
        $eshopQuery->setLimit(10);
  
        // 6. Add sorting option by relevance
        $eshopQuery->setSortCriteria(array(new RelevanceSorting()));
  
        // 7. Add product facet option
        $productFacet = new SimpleProductFieldFacet(
            array(
                'fieldName' => 'manufacturer',
                'labelKey' => 'manufacturer',
                'filterType' => 'multi_or',
                'sortType' => 'alpha',
            )
        );
        $eshopQuery->addFacet($productFacet);
        

>         // 8. Finally we perform the search.

Reviewer: ```suggestion
        // 8. Perform the search
```

        $productSearchResult = $searchService->searchProducts($eshopQuery, new SearchContext());
  

>         // 9. The hits could be found here

Reviewer: ```suggestion
        // 9. The hits can be found here
```

        $productSearchResult->facets
    }
}
```

### Conditions API

`ContentTypesCondition` - filters results by Content Type. Valid Content Types are for example `ses_product`, `ses_category` or `ses_content`.

`SearchTermCondition` - filters results by phrase, potentially in a specific field. For example: search for `SE10000` in field `sku`:

``` php
$myQuery->addCondition(new SearchTermCondition(array(
    'searchTerm' => 'SE10000',
    'fieldRestrictions' => 'ses_product_ses_sku_value_s'
)));
```

`SectionCondition` - filters results by Section ID. Returns only elements that belong to the given Section.

`SubtreeCondition` - filters results by setting a path. Only elements under this path are fetched.

`VisibilityCondition` - filters results that are shown or hidden.

### Boosting API

`FieldBoosting` - defines which fields should be boosted in search result. They have higher priority in search.

### Sorting API

`RelevanceSorting` - sorts the results by relevance, which is internal Solr implementation for the `score` field.

`ProductFieldSorting` - sorting for products. Supports sorting by name, SKU or price in a given direction.

> `ContentNameSorting` - sorting by content Field name in a given direction.

Reviewer: ```suggestion
`ContentNameSorting` - sorts the results by content Field name in a given direction.
```
